globus_cli/commands/endpoint/create.py

```diff
@@ -25,32 +27,49 @@
 @click.option('--globus-connect-personal', 'endpoint_type',
               flag_value='personal', default=True, show_default=True,
               help='This endpoint is a Globus Connect Personal endpoint')
-def endpoint_create(endpoint_type, display_name, description, organization,
-                    department, keywords, contact_email, contact_info,
-                    info_link, public, default_directory, force_encryption,
-                    oauth_server, myproxy_server, myproxy_dn):
+@click.option("--shared", default=None, type=ENDPOINT_PLUS_REQPATH,
+              metavar=ENDPOINT_PLUS_REQPATH.metavar,
+              help=("This endpoint is a shared endpoint hosted "
+                    "on the given endpoint and path"))
+def endpoint_create(endpoint_type, shared, display_name, description,
+                    info_link, contact_info, contact_email, organization,
+                    department, keywords, public, location, disable_verify,
+                    myproxy_dn, myproxy_server, oauth_server, force_encryption,
+                    default_directory, subscription_id, network_use,
+                    max_concurrency, preferred_concurrency,
+                    max_parallelism, preferred_parallelism):
     """
     Executor for `globus endpoint create`
     """
+    client = get_client()
 
-    # omit the `is_globus_connect` key if not GCP, otherwise include as `True`
-    is_globus_connect = endpoint_type == 'personal' or None
-    ep_doc = assemble_generic_doc(
-        'endpoint',
-        is_globus_connect=is_globus_connect,
-        display_name=display_name, description=description,
-        organization=organization, department=department,
-        keywords=keywords, contact_email=contact_email,
-        contact_info=contact_info, info_link=info_link,
-        force_encryption=force_encryption, public=public,
-        default_directory=default_directory,
-        myproxy_server=myproxy_server, myproxy_dn=myproxy_dn,
-        oauth_server=oauth_server)
+    # validate params
+    if shared:
+        endpoint_type = "shared"
+    args, _, _, values = inspect.getargvalues(inspect.currentframe())
+    params = dict((k, values[k]) for k in args)
```

Comment:
I think ran into some issue using **kwargs on my first iteration of this and switched to inspect as a workaround, given that I can't remember what the issue was and this has gotten much cleaner since then I'll switch back to **kwargs.